src/app/index.ts

```diff
@@ -53,10 +51,22 @@ declare const UIJarConfiguration: AppConfig;
         AppComponent
     ],
     providers: [
-        { provide: 'AppData', useFactory: generatedOutput.getAppData },
-        { provide: 'AppConfig', useValue: typeof UIJarConfiguration !== 'undefined' ? UIJarConfiguration : {} }
-    ]
-})
-export class UIJarModule {
+        { provide: 'AppData', useFactory: generatedOutput.getAppData }
+    ],
+    entryComponents: [],
+    schemas: []
+};
+
+export const UIJarModule = (settings: AppConfig = { config: {} }) => {
```

Comment:
@nording not sure what's the correct approach, since doing that we will fuck up the AOT build. Do you have any suggestion?

Reply:
Not sure either, I will have to digg a bit!